src/main/java/org/primefaces/context/PrimeRequestContext.java

```diff
@@ -207,22 +208,25 @@ public void release() {
      * @return if secure or not.
      */
     public boolean isSecure() {
-        // currently called once per request - later we might cache the result per request
-        // and even the method lookup
-        Object request = context.getExternalContext().getRequest();
+        // currently called once per request
```

Comment:
This comment is no longer true now you are caching it after the first result.  @tandraschko I think this is OK because i doubt any site has a mix of HTTP and HTTPS.  At least they should not!